space_packet_parser/packets.py

```diff
@@ -369,7 +420,37 @@ def ccsds_generator(
         packet_bytes = read_buffer[current_pos:current_pos + n_bytes_packet]
         current_pos += n_bytes_packet
         # Wrap the bytes in a RawPacketData object that adds convenience methods for parsing the header
-        yield RawPacketData(packet_bytes)
+        ccsds_packet = CCSDSPacketBytes(packet_bytes)
+
+        if not combine_segmented_packets or ccsds_packet.sequence_flags == SequenceFlags.UNSEGMENTED:
+            yield ccsds_packet
+        elif ccsds_packet.sequence_flags == SequenceFlags.FIRST:
+            _segmented_packets[ccsds_packet.apid] = [ccsds_packet]
+            continue
+        elif not _segmented_packets.get(ccsds_packet.apid, []):
+            warnings.warn("Continuation packet found without declaring the start, skipping this packet.")
+            continue
+        elif ccsds_packet.sequence_flags == SequenceFlags.CONTINUATION:
+            _segmented_packets[ccsds_packet.apid].append(ccsds_packet)
+            continue
+        else:  # raw_packet_data.sequence_flags == packets.SequenceFlags.LAST:
+            _segmented_packets[ccsds_packet.apid].append(ccsds_packet)
+            # We have received the final packet, close it up and combine all of
+            # the segmented packets into a single "packet" for XTCE parsing
+            sequence_counts = [p.sequence_count for p in _segmented_packets[ccsds_packet.apid]]
+            if not all((sequence_counts[i + 1] - sequence_counts[i]) % 16384 == 1
+                        for i in range(len(sequence_counts) - 1)):
+                warnings.warn(f"Continuation packets for apid {ccsds_packet.apid} "
+                              f"are not in sequence {sequence_counts}, skipping these packets.")
```

Comment:
I think it would be safe to just reorder these according the sequence counter, but maybe leave the warning even if we can fix it because it's probably not expected that the packets are out of order.

Reply:
This accounts for gaps though too, so it isn't just re-ordering.